templates/skeleton/app/components/Header.tsx

```diff
@@ -159,15 +163,18 @@ function CartToggle({cart}: Pick<HeaderProps, 'cart'>) {
   return (
     <Suspense fallback={<CartBadge count={null} />}>
       <Await resolve={cart}>
-        {(cart) => {
-          if (!cart) return <CartBadge count={0} />;
-          return <CartBadge count={cart.totalQuantity || 0} />;
-        }}
+        <CartBanner />
       </Await>
     </Suspense>
   );
 }
 
+function CartBanner() {
+  const originalCart = useAsyncValue() as CartApiQueryFragment | null;
+  const cart = useOptimisticCart(originalCart);
+  return <CartBadge count={cart?.totalQuantity ?? null} />;
```

Comment:
that `null` should be just `0`. The suspense already took care of when data is not ready (which is why it is `null`).

This `CartBadge` should display `0` since we have resolved the cart query by this point.

Reply:
Thanks for the explanation, I fixed it!